### PR TITLE
Display self-revive cooldown in unconscious state

### PR DESCRIPTION
**Pull request by blackwaterbread**
## What type of PR is this?
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

We need a UI that tells us how long we have left to cooldown Self-Revive when in unconscious.

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

Makes the player in unconscious, and verify that the cooldown UI is displayed in the top right corner.

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/functions/Revive/fn_unconscious.sqf [hunk 1]
```diff
@@ -16,6 +17,41 @@ private _fnc_applyPostEffect = {
 	"filmGrain" ppEffectEnable true;
 };
 
+private _fnc_selfReviveCountdownStart = {
+	private _diff = (player getVariable ["A3A_selfReviveTimeout", -1]) - time;
```
Comment (Silence-112) [inline A3A/addons/core/functions/Revive/fn_unconscious.sqf]:
```suggestion
	private _diff = (_unit getVariable ["A3A_selfReviveTimeout", -1]) - time;
```

Use the _unit variable for consistency

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Make sure to read the suggestion. Other than that, looks fine

**Review by Silence-112 — COMMENTED** [review]
The hint doesn't disappear when you hit R to respawn and the cooldown keeps ticking even after respawning. Is that intended behaviour?

Using withstand does seem to hide the hint as usual, that or it gets overwritten by the antistasi hint.

You should probably be using the `A3A_fnc_customHint` function instead of `hintSilent` as using normal hints can cause them to be overwritten without any chance to read them.

**Comment by blackwaterbread** [comment]
Thank you for your detailed review. Working on it based on your suggestions.

**Review by Silence-112 — COMMENTED** [review]
Make sure to pass a silent notification argument to the customHint call, else you'll be spamming people with sounds

`["FooBar", "Hello World", true] call A3A_fnc_customHint;`

Other than that, it seems fine now

**Review by Silence-112 — APPROVED** [review]
Looks good to me, will need to be tested by us before merge

**Comment by blackwaterbread** [comment]
Great, let me know if there's anything else need to work on

**Pull request merged**